packages/ia-components/sandbox/details/CrawlConfig.js

```diff
@@ -0,0 +1,142 @@
+import ReactDOM from "react-dom";
+
+const debug = require('debug')('dweb-archive:CrawlConfig');
+import canonicaljson from '@stratumn/canonicaljson';
+import React from "react";
+import IAReactComponent from '../IAReactComponent';
+import {gatewayServer} from '../../util.js';
+//DwebTransports is not needed, its a global
+
+/**
+ * Component used in a tile bar on dweb-mirror to display and control Crawl functionality.
+ * It renders a button which indicates whether the viewed item is being crawled, and info about that crawl,
+ * and allows changing the crawl status.
+ *
+ * Note this component is under development (May 2019) and will likely change to be more functional, which is why there is
+ * commented out code there.
+ *
+ * Behavior:
+ * On construction: it sets an instance variable on the class so other parts of the UI can send info to it.
+ *
+ * On Clicking:
+ *   It cycles the "level" through none / "details" / "all"
+ *   It rerenders, showing new level
+ *   It sends the new level to the server.
+ *
+ * <CrawlConfig
+ *  identifier  of item
+ *  level       Current crawling level of object
+ *  search      Current search parameters for crawl
+ *  downloaded  true if item downloaded to details level (displayed as white if not crawling)
+ *  />
+ *
+ */
+
+
+
+export default class CrawlConfig extends IAReactComponent {
+    /* -- Not used with ReactFake yet
+    static propTypes = {
+        identifier: PropTypes.string,
+        level: PropTypes.string,
+        search: PropTypes.object,
+        downloaded: ProbTypes.boolean
+    };
+    */
+    constructor(props)
+    {
+        super(props);   // { identifier, level, search, downloaded }
+        this.setState(props);
+        CrawlConfig.instance = this; // Allow finding it
+    }
+
+    loaded() {
+        return (typeof this.state.level !== "undefined");
+    }
+    static findAndSetState(state) {
+        this.instance.setState(state);
+    }
+    render() {
+        //TODO-CONFIG make it editable
+        if ((typeof DwebArchive === "undefined") || (DwebArchive.mirror === null))  {
+            return null
+        } else {
```

Comment:
Rejected: There are opinions as to whether or not its better to separate branches with else, it makes zero functional difference so I'm not sure why you are delaying this push because of something like this.